pkg/metrics/client_go_adapter.go

```diff
@@ -0,0 +1,299 @@
+		Help:      "How long an API list takes to return and decode for the reflectors",
+	}, []string{"name"})
+
+	itemsPerList = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Subsystem: reflectorSubsystem,
+		Name:      "items_per_list",
+		Help:      "How many items an API list returns to the reflectors",
+	}, []string{"name"})
+
+	watchesTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Subsystem: reflectorSubsystem,
+		Name:      "watches_total",
+		Help:      "Total number of API watches done by the reflectors",
+	}, []string{"name"})
+
+	shortWatchesTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Subsystem: reflectorSubsystem,
+		Name:      "short_watches_total",
+		Help:      "Total number of short API watches done by the reflectors",
+	}, []string{"name"})
+
+	watchDuration = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Subsystem: reflectorSubsystem,
+		Name:      "watch_duration_seconds",
+		Help:      "How long an API watch takes to return and decode for the reflectors",
+	}, []string{"name"})
+
+	itemsPerWatch = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Subsystem: reflectorSubsystem,
+		Name:      "items_per_watch",
+		Help:      "How many items an API watch returns to the reflectors",
+	}, []string{"name"})
+
+	lastResourceVersion = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: reflectorSubsystem,
+		Name:      "last_resource_version",
+		Help:      "Last resource version seen for the reflectors",
+	}, []string{"name"})
+
+	// workqueue metrics
+
+	workQueueSubsystem = "workqueue"
+
+	depth = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem:   workQueueSubsystem,
+		Name:        "depth",
+		Help:        "Current depth of workqueue",
+	}, []string{"name"})
+
+	adds = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Subsystem:   workQueueSubsystem,
+		Name:        "adds_total",
+		Help:        "Total number of adds handled by workqueue",
+	}, []string{"name"})
+
+	latency = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Subsystem:   workQueueSubsystem,
+		Name:        "queue_latency_seconds",
+		Help:        "How long in seconds an item stays in workqueue before being requested.",
+		Buckets:     prometheus.ExponentialBuckets(10e-9, 10, 10),
+	}, []string{"name"})
+
+	workDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Subsystem:   workQueueSubsystem,
+		Name:        "work_duration_seconds",
+		Help:        "How long in seconds processing an item from workqueue takes.",
+		Buckets:     prometheus.ExponentialBuckets(10e-9, 10, 10),
+	}, []string{"name"})
+
+	retries = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Subsystem:   workQueueSubsystem,
+		Name:        "retries_total",
+		Help:        "Total number of retries handled by workqueue",
+	}, []string{"name"})
+
+	longestRunning = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: workQueueSubsystem,
+		Name:      "longest_running_processor_microseconds",
```

Comment:
microseconds?
It seems this is not following the convention.

Reply:
yeah, but the current code requires it atm.  it's fixed in future releases